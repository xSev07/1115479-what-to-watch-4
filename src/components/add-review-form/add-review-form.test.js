import React from "react";
import renderer from "react-test-renderer";
import AddReviewForm from "./add-review-form";

describe(`Should AddReviewForm render correctly`, () => {
  it(`should render basic`, () => {
    const tree = renderer
      .create(
          <AddReviewForm
            isSubmitDisabled={false}
            isCommentSending={false}
            commentSendingError={false}
            onSubmit={() => {}}
            onChange={() => {}}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render with submit disabled`, () => {
    const tree = renderer
      .create(
          <AddReviewForm
            isSubmitDisabled={true}
            isCommentSending={false}
            commentSendingError={false}
            onSubmit={() => {}}
            onChange={() => {}}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render with sending error`, () => {
    const tree = renderer
      .create(
          <AddReviewForm
            isSubmitDisabled={false}
            isCommentSending={false}
            commentSendingError={true}
            onSubmit={() => {}}
            onChange={() => {}}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
