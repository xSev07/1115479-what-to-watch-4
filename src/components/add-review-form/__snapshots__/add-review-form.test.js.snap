// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should AddReviewForm render correctly should render basic 1`] = `
<form
  action="#"
  className="add-review__form"
  disabled={false}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <div
    className="rating"
  >
    <div
      className="rating__stars"
    >
      <input
        checked={false}
        className="rating__input"
        id="star-1"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={1}
      />
      <label
        className="rating__label"
        htmlFor="star-1"
      >
        Rating 
        1
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-2"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={2}
      />
      <label
        className="rating__label"
        htmlFor="star-2"
      >
        Rating 
        2
      </label>
      <input
        checked={true}
        className="rating__input"
        id="star-3"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={3}
      />
      <label
        className="rating__label"
        htmlFor="star-3"
      >
        Rating 
        3
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-4"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={4}
      />
      <label
        className="rating__label"
        htmlFor="star-4"
      >
        Rating 
        4
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-5"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={5}
      />
      <label
        className="rating__label"
        htmlFor="star-5"
      >
        Rating 
        5
      </label>
    </div>
  </div>
  <div
    className="add-review__text"
  >
    <textarea
      className="add-review__textarea"
      id="review-text"
      name="text"
      onChange={[Function]}
      placeholder="Review text"
      value=""
    />
    <div
      className="add-review__submit"
    >
      <button
        className="add-review__btn"
        disabled={true}
        onClick={[Function]}
        style={
          Object {
            "opacity": 0.5,
          }
        }
        type="submit"
      >
        Post
      </button>
    </div>
  </div>
</form>
`;

exports[`Should AddReviewForm render correctly should render with sending error 1`] = `
<form
  action="#"
  className="add-review__form"
  disabled={false}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <div>
    <p>
      Error sending comment
    </p>
  </div>
  <div
    className="rating"
  >
    <div
      className="rating__stars"
    >
      <input
        checked={false}
        className="rating__input"
        id="star-1"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={1}
      />
      <label
        className="rating__label"
        htmlFor="star-1"
      >
        Rating 
        1
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-2"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={2}
      />
      <label
        className="rating__label"
        htmlFor="star-2"
      >
        Rating 
        2
      </label>
      <input
        checked={true}
        className="rating__input"
        id="star-3"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={3}
      />
      <label
        className="rating__label"
        htmlFor="star-3"
      >
        Rating 
        3
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-4"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={4}
      />
      <label
        className="rating__label"
        htmlFor="star-4"
      >
        Rating 
        4
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-5"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={5}
      />
      <label
        className="rating__label"
        htmlFor="star-5"
      >
        Rating 
        5
      </label>
    </div>
  </div>
  <div
    className="add-review__text"
  >
    <textarea
      className="add-review__textarea"
      id="review-text"
      name="text"
      onChange={[Function]}
      placeholder="Review text"
      value=""
    />
    <div
      className="add-review__submit"
    >
      <button
        className="add-review__btn"
        disabled={true}
        onClick={[Function]}
        style={
          Object {
            "opacity": 0.5,
          }
        }
        type="submit"
      >
        Post
      </button>
    </div>
  </div>
</form>
`;

exports[`Should AddReviewForm render correctly should render with submit disabled 1`] = `
<form
  action="#"
  className="add-review__form"
  disabled={false}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <div
    className="rating"
  >
    <div
      className="rating__stars"
    >
      <input
        checked={false}
        className="rating__input"
        id="star-1"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={1}
      />
      <label
        className="rating__label"
        htmlFor="star-1"
      >
        Rating 
        1
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-2"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={2}
      />
      <label
        className="rating__label"
        htmlFor="star-2"
      >
        Rating 
        2
      </label>
      <input
        checked={true}
        className="rating__input"
        id="star-3"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={3}
      />
      <label
        className="rating__label"
        htmlFor="star-3"
      >
        Rating 
        3
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-4"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={4}
      />
      <label
        className="rating__label"
        htmlFor="star-4"
      >
        Rating 
        4
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-5"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={5}
      />
      <label
        className="rating__label"
        htmlFor="star-5"
      >
        Rating 
        5
      </label>
    </div>
  </div>
  <div
    className="add-review__text"
  >
    <textarea
      className="add-review__textarea"
      id="review-text"
      name="text"
      onChange={[Function]}
      placeholder="Review text"
      value=""
    />
    <div
      className="add-review__submit"
    >
      <button
        className="add-review__btn"
        disabled={true}
        onClick={[Function]}
        style={
          Object {
            "opacity": 0.5,
          }
        }
        type="submit"
      >
        Post
      </button>
    </div>
  </div>
</form>
`;
